File: src/main/java/com/pedigree/ui/JavaFxGraphicsContext.java
package com.pedigree.ui;

import com.pedigree.render.GraphicsContext;
import javafx.scene.paint.Color;
import javafx.scene.text.Font;

public class JavaFxGraphicsContext implements GraphicsContext {

    private final javafx.scene.canvas.GraphicsContext gc;

    public JavaFxGraphicsContext(javafx.scene.canvas.GraphicsContext gc) {
        this.gc = gc;
    }

    @Override
    public void setStrokeColor(int r, int g, int b, double a) {
        gc.setStroke(Color.rgb(r, g, b, a));
    }

    @Override
    public void setFillColor(int r, int g, int b, double a) {
        gc.setFill(Color.rgb(r, g, b, a));
    }

    @Override
    public void setLineWidth(double w) {
        gc.setLineWidth(w);
    }

    @Override
    public void setFontSize(double size) {
        try {
            gc.setFont(Font.font(size));
        } catch (Throwable ignore) {
            // Fallback: ignore if font cannot be set in current context
        }
    }

    @Override
    public void drawLine(double x1, double y1, double x2, double y2) {
        gc.strokeLine(x1, y1, x2, y2);
    }

    @Override
    public void fillRect(double x, double y, double w, double h) {
        gc.fillRect(x, y, w, h);
    }

    @Override
    public void drawRect(double x, double y, double w, double h) {
        gc.strokeRect(x, y, w, h);
    }

    @Override
    public void drawText(String text, double x, double y) {
        gc.fillText(text, x, y);
    }
}

File: src/main/kotlin/com/family/tree/ui/JavaFxGraphicsContext.kt
package com.family.tree.ui

import com.family.tree.render.GraphicsContext
import javafx.scene.paint.Color
import javafx.scene.text.Font

class JavaFxGraphicsContext(private val gc: javafx.scene.canvas.GraphicsContext) : GraphicsContext {
    override fun setStrokeColor(r: Int, g: Int, b: Int, a: Double) {
        gc.stroke = Color.rgb(r, g, b, a)
    }

    override fun setFillColor(r: Int, g: Int, b: Int, a: Double) {
        gc.fill = Color.rgb(r, g, b, a)
    }

    override fun setLineWidth(w: Double) {
        gc.lineWidth = w
    }

    override fun setFontSize(size: Double) {
        try {
            gc.font = Font.font(size)
        } catch (_: Throwable) {
            // ignore if font cannot be set in current context
        }
    }

    override fun drawLine(x1: Double, y1: Double, x2: Double, y2: Double) {
        gc.strokeLine(x1, y1, x2, y2)
    }

    override fun fillRect(x: Double, y: Double, w: Double, h: Double) {
        gc.fillRect(x, y, w, h)
    }

    override fun drawRect(x: Double, y: Double, w: Double, h: Double) {
        gc.strokeRect(x, y, w, h)
    }

    override fun drawText(text: String, x: Double, y: Double) {
        gc.fillText(text, x, y)
    }

    override fun measureTextWidth(text: String): Double {
        if (text.isEmpty()) return 0.0
        val textNode = javafx.scene.text.Text(text)
        textNode.font = gc.font
        return textNode.boundsInLocal.width
    }
}
